bftengine/src/bftengine/AssortedInternalMsgCloner.hpp

```diff
@@ -0,0 +1,29 @@
+// Concord
+//
+// Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").  You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include "messages/InternalMessage.hpp"
+#include "messages/AssortedInternalMsg.hpp"
+
+namespace bftEngine::impl {
+class AssortedInternalMsgCloner {
+ public:
+  static InternalMessage getCloneOfViewChangeIndicatorInternalMsg(ReplicaAsksToLeaveViewMsg::Reason viewChangeReason) {
```

Comment:
You actually not cloning any existing message, but creating a new one. Please rename the functions.
But I would actually get rid of this class at all. You could create the messages inline when needed.
There is no logic here.

Reply:
Thanks @yuliasherman for your comment, I have incorporated the changes.